### PR TITLE
Move containerization bits into rosindex

### LINKED ISSUE #23: Move containerization bits into rosindex
In our quest to get rid of submodules, we'd like to move all rosindex containerization specific bits, currently living at [ros-infrastructure/rosindex_containers](https://github.com/ros-infrastructure/rosindex_containers), into rosindex repo itself. Preferably in an appropriately named subdirectory.

### PR DESCRIPTION
**Pull request by apojomovsky**
Fixes #23 
- Moves the Dockerfile as well as the helper scripts into the `rosindex/docker` directory.
- Improved dockerfile a bit to reduce final image size
- Improved helper scripts to make them agnostic to the path from which they've been called.

### CODE REVIEW

File: docker/build.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+BASE_DIR=`dirname $( readlink -m $( type -p $0 ))`
```
Comment (hidmic) [inline docker/build.sh]: @apojomovsky nit: consider `SCRIPT_DIR` as IMHO it's more descriptive.

File: docker/run.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+BASE_DIR=`dirname $( readlink -m $( type -p $0 ))`
+
+# Ensure your SSH private key has been added to the ssh-agent as follows:
+# ssh-add /home/$USER/.ssh/id_rsa
+
+# Example Usage:
+# ./run.sh make build
+
+docker run\
```
Comment (hidmic) [inline docker/run.sh]: @apojomovsky wish: consider jumping into bash if no argument is passed to `run.sh`.

**Review by hidmic — APPROVED** [review]
@apojomovsky Left just a few silly comments. LGTM!

File: docker/build.sh (hunk 1 above)
Comment (apojomovsky) [reply docker/build.sh]: agree with you, done!

File: docker/run.sh (hunk 2 above)
Comment (apojomovsky) [reply docker/run.sh]: you're right, done!

**Comment by apojomovsky** [comment]
Thanks for the review @hidmic , just finished addressing your comments. Could you PTAL?

**Comment by apojomovsky** [comment]
Oh, nevermind, just saw your "LGTM" so I'm merging this now (:

**Pull request merged**